clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -341,7 +349,8 @@ private boolean handleKafkaRequest(byte[] requestBytes) throws IOException, Auth
 
             LOG.debug("Handling Kafka request {}", apiKey);
 
-            RequestContext requestContext = new RequestContext(header, connectionId, clientAddress,
+
+            RequestContext requestContext = new RequestContext(header, connectionId, clientAddress(),
```

Comment:
Do we want to retrieve the client address from the transport layer for every request? It should always be the same after the connection is established right?

Reply:
Yeah, didn't think it was a big deal since this is just handling the pre-authenticated requests, but I can add a field instead.

Reply:
That's fair, we can leave as is.